echopop/extensions/diagnostics.py

```diff
@@ -0,0 +1,510 @@
+
+        # Get the full datasets
+        # ---- Get stratification information
+        strata_info = self.data.input["spatial"]["inpfc_strata_df"].copy()
+        # ---- Initialize
+        DATAINPUT = {"Biomass": {}, "Biomass density": {}}
+        # ---- Kriging
+        if "kriging" in self.data.results:
+            strat_krig = self.data.results["stratified"]["kriging"].copy()
+            # ---- Add 'biomass'
+            DATAINPUT["Biomass"].update(
+                {
+                    "kriging": pd.DataFrame(
+                        {
+                            "stratum": strata_info.stratum_inpfc,
+                            "total": strat_krig["estimate"]["strata"]["total"] * 1e-6,
+                            "lower": [k[0] * 1e-6 for k in strat_krig["ci"]["strata"]["total"]],
+                            "upper": [k[1] * 1e-6 for k in strat_krig["ci"]["strata"]["total"]],
+                        }
+                    )
+                }
+            )
+            # ---- Add 'biomass density'
+            DATAINPUT["Biomass density"].update(
+                {
+                    "kriging": pd.DataFrame(
+                        {
+                            "stratum": strata_info.stratum_inpfc,
+                            "density": strat_krig["estimate"]["strata"]["density"] * 1e-6,
+                            "lower": [k[0] * 1e-6 for k in strat_krig["ci"]["strata"]["density"]],
+                            "upper": [k[1] * 1e-6 for k in strat_krig["ci"]["strata"]["density"]],
+                        }
+                    )
+                }
+            )
+        # ---- Transect
+        if "transect" in self.data.results:
+            strat_krig = self.data.results["stratified"]["transect"].copy()
+            # ---- Add 'biomass'
+            DATAINPUT["Biomass"].update(
+                {
+                    "transect": pd.DataFrame(
+                        {
+                            "stratum": strata_info.stratum_inpfc,
+                            "total": strat_krig["estimate"]["strata"]["total"] * 1e-6,
+                            "lower": [k[0] * 1e-6 for k in strat_krig["ci"]["strata"]["total"]],
+                            "upper": [k[1] * 1e-6 for k in strat_krig["ci"]["strata"]["total"]],
+                        }
+                    )
+                }
+            )
+            # ---- Add 'biomass density'
+            DATAINPUT["Biomass density"].update(
+                {
+                    "transect": pd.DataFrame(
+                        {
+                            "stratum": strata_info.stratum_inpfc,
+                            "density": strat_krig["estimate"]["strata"]["density"] * 1e-6,
+                            "lower": [k[0] * 1e-6 for k in strat_krig["ci"]["strata"]["density"]],
+                            "upper": [k[1] * 1e-6 for k in strat_krig["ci"]["strata"]["density"]],
+                        }
+                    )
+                }
+            )
```

Comment:
You can use a for loop to set both "kriging" and "transect" entries:
```
for agg_type in ["kriging", "transect"]:
  if agg_type in self.data.results:
   ...
```
Since the majority of the code is identical, this will make it more robust to potential typos.

Reply:
Along the same line, if you use the same name for the quantity, you don't need to separately handle `var_columns = {"Biomass": "total", "Biomass density": "density"}` in the plotting

Reply:
> Along the same line, if you use the same name for the quantity, you don't need to separately handle var_columns = {"Biomass": "total", "Biomass density": "density"} in the plotting below, as that's the only difference between the biomass and biomass density plots.

I'm not sure I fully follow this comment/suggestion. The `"total"` and `"density"` results are stored in separate dictionaries under both `"transect"` and `"kriging"`, so I'm not sure how using the same name would be viable here?

Reply:
Oh, unless you are referring to how the column for the values for the DataFrame within `DATAINPUT` is named. In that case, I totally agree that using a shared column name like `"mean"` would make more sense.

Reply:
Yes, column names were what I meant, since then it wouldn't require separate handling.